packages/Babylonian-Printbugger.package/BPSwimlane.class/instance/newLabelValuePairMorphFor..st

```diff
@@ -1,6 +1,22 @@
 private - ui
 newLabelValuePairMorphFor: aTraceValue
 
+	| labelMorph valueMorph isTraceValueVisible |
+	isTraceValueVisible := self isTraceValueVisible: aTraceValue.
+	labelMorph := BPLabelMorph 
+					newFor: aTraceValue 
+					withButton: (self newToggleAnnotationVisibilityButtonFor: aTraceValue isVisible: isTraceValueVisible).
+	valueMorph := self newValueMorphFor: aTraceValue.
+	
+	isTraceValueVisible 
+		ifFalse: [
+			valueMorph hide.
+			labelMorph downsized].
+	
+	"flash the label if the belonging annotation just got its visibility toggled in the printbugger"	
+	((annotationIdToFlash ~= nil) and: (annotationIdToFlash = aTraceValue belongingAnnotationId)) 
```

Comment:
this logic feels out of place here. besides the method being too long, i would have expected the valuemorph itself being responsible for displaying or hiding itself. maybe an own class would start to make sense here. same for `newToggleAnnotationVisibilityButtonFor: aTraceValue  isVisible: aBoolean`

Reply:
This might be the point where we could extract a `BPPrintbuggerLineMorph` or `BPPrintbuggerLabelValuePairMorph`.

Reply:
with the other open mr this is probably a good idea